### ISSUE TITLE
Python box is not working for df in katana environment (it is working fine in simple LK)

### ISSUE THREAD

**Issue by mszel** [lynxkite/lynxkite#298]
The following python box is working well in LK:

- **inputs**: df.label, df.score
- **outputs**: df.fpr: float, df.tpr: float
- **Python code**: 
from sklearn.metrics import roc_curve
fpr, tpr, _ = roc_curve(df.label, df.score)
df = pd.DataFrame({'fpr': fpr, 'tpr': tpr})

After copying the box to Katana (running on the exact same input), it is not working (neither is the simplest query).

**Comment by mszel** [comment]
The error I got:

Failed to load table.
Unable to infer schema for Parquet. It must be specified manually.

**Comment by darabos** [comment]
If it's working on one LynxKite instance but not on another, it's probably a configuration issue. @JMLizano, do you have any insight on how the two instances may differ?

I may be able to tell you more if you give me a stack trace or a pointer to the instance with the problem. Thanks!

**Comment by darabos** [comment]
To comment on the Katana part: The Python box is unaffected by Katana, so I don't think the Katana integration itself would be the issue. But I suspect you're also running different LynxKite builds between the two systems. Can you maybe point me to which commit each instance is built from? Thanks!

**Comment by JMLizano** [comment]
Hi Dani, thanks for looking into this.

This this [the configuration ](https://github.com/lynxkite/lynxkite-katana/blob/main/databricks/lynxkite-katana-on-databricks.json)that I used to set up the cluster.

My guess is that the LK version that I am using (the JAR at dbfs:/lynxkite-build/lynxkite-0.1-SNAPSHOT-DATABRICKS.jar) has some differences with the current LK version. This jar was generated on July 28th, but I don't know how to check the specific version of LK.

Maybe the easiest is to just generate and up to date build and check if that solves the issue.

**Comment by darabos** [comment]
Thanks! If I had a repro, I could look into this.

**Comment by JMLizano** [comment]
Sure, this is the link to the cluster:

https://dbc-1a31834b-8f7f.cloud.databricks.com/?o=622560398774279#setting/clusters/1102-054927-kw0k5lay/configuration

and this is the link to the workspace:

https://dbc-dp-622560398774279.cloud.databricks.com/driver-proxy/o/622560398774279/1102-054927-kw0k5lay/2200/#/workspace/article_dev_ERocheteau_opp/uc_01_1disease_outcomes

The Python box is at the top. It seems to fail when you try to connect it to any table output (I tried several).

**Comment by darabos** [comment]
Perfect, thanks! For posterity, here's the stack trace:

```
org.apache.spark.sql.AnalysisException: Unable to infer schema for Parquet. It must be specified manually.
	at org.apache.spark.sql.errors.QueryCompilationErrors$.dataSchemaNotSpecifiedError(QueryCompilationErrors.scala:1160)
	at org.apache.spark.sql.execution.datasources.DataSource.$anonfun$getOrInferFileFormatSchema$13(DataSource.scala:236)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.execution.datasources.DataSource.getOrInferFileFormatSchema(DataSource.scala:236)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:460)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:368)
	at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:324)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:324)
	at org.apache.spark.sql.DataFrameReader.parquet(DataFrameReader.scala:752)
	at org.apache.spark.sql.DataFrameReader.parquet(DataFrameReader.scala:737)
	at com.lynxanalytics.biggraph.graph_api.SparkDomain.$anonfun$relocateFrom$15(SparkDomain.scala:466)
	at com.lynxanalytics.biggraph.graph_api.SafeFuture$.wrapException(SafeFuture.scala:29)
	at com.lynxanalytics.biggraph.graph_api.SafeFuture$.$anonfun$async$1(SafeFuture.scala:14)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
	Suppressed: java.lang.Throwable: SafeFuture creator's stack trace
		at com.lynxanalytics.biggraph.graph_api.SafeFuture$.com$lynxanalytics$biggraph$graph_api$SafeFuture$$unwrapException(SafeFuture.scala:43)
		at com.lynxanalytics.biggraph.graph_api.SafeFuture$.async(SafeFuture.scala:14)
		at com.lynxanalytics.biggraph.graph_api.SparkDomain.relocateFrom(SparkDomain.scala:466)
		at com.lynxanalytics.biggraph.graph_api.DataManager.$anonfun$ensureThenRelocate$1(DataManager.scala:153)
		at com.lynxanalytics.biggraph.graph_api.SafeFuture.$anonfun$flatMap$1(SafeFuture.scala:59)
		at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
		at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
		... 4 more
```

**Comment by darabos** [comment]
Sorry, I don't think I can fix this quickly. I think the key difference between the good and the bad case is that in the bad case you're running on Databricks. https://stackoverflow.com/questions/44954892/unable-to-infer-schema-when-loading-parquet-file says you typically get this error if you try to read an empty directory as Parquet. This probably happens due to an issue with passing data between Spark and Sphynx.

I'll set up a better test environment for investigating this on Monday.

If "Compute in Python" for graphs works, maybe you can use that as a workaround until we have a proper fix.

**Comment by JMLizano** [comment]
Thanks Dani! @mszel is the workaround good for you?

**Comment by mszel** [comment]
yes, it was good :). Sorry, I missed this chat.

**Comment by darabos** [comment]
Sorry, I didn't investigate this further in the end because the workaround worked. We'll look into it when it bites us next time.
